pkg/fs/jsonfile.go

```diff
@@ -0,0 +1,44 @@
+package fs
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+// MarshalJSONFile writes the JSON encoding of v to a file named by filename.
+// The file is created atomically. If the file does not exist, Marshal creates
+// it with permissions perm.  Internally MarshalJSONFile uses json.Marshal.
+func MarshalJSONFile(v interface{}, filename string, perm os.FileMode) error {
+	tmppath := filename + ".tmp"
+	f, err := OpenFile(tmppath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, perm)
```

Comment:
We need O_EXCL here to prevent one goroutine from stomping on another.

Reply:
How comes the stdlib os.Open does not pass this flag?

Reply:
os.Open doesn't promise exclusive access to the file.

I think the right point of comparison is io/ioutil.TempFile, whose documentation says, "Multiple programs calling TempFile simultaneously will not choose the same file." TempFile is very O_EXCL (https://github.com/golang/go/blob/go1.14.4/src/io/ioutil/tempfile.go#L60).

Reply:
Good call, thanks.